### PR TITLE
Update the build process to use toltecmk

### LINKED ISSUE #650: Unable to build packages `assert next_token == "("`
When trying to build any toltec package I am seeing this error:

```shell
$  make koreader
./scripts/package_build.py  "koreader"
Traceback (most recent call last):
  File "/var/home/alistair/sources/remarkable/toltec/./scripts/package_build.py", line 44, in <module>
    repo = Repo(paths.RECIPE_DIR, paths.REPO_DIR)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/alistair/sources/remarkable/toltec/scripts/toltec/repo.py", line 56, in __init__
    self.generic_recipes[name] = GenericRecipe.from_file(
                                 ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/alistair/sources/remarkable/toltec/scripts/toltec/recipe.py", line 45, in from_file
    return GenericRecipe(name, path, recipe.read())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/alistair/sources/remarkable/toltec/scripts/toltec/recipe.py", line 58, in __init__
    variables, functions = bash.get_declarations(definition)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/alistair/sources/remarkable/toltec/scripts/toltec/bash.py", line 147, in get_declarations
    assert next_token == "("
AssertionError
make: *** [Makefile:60: koreader] Error 1

```

It doesn't matter what package I build, I see the same error.

```shell
$ python3 --version
Python 3.11.0
```

```shell
bash --version
GNU bash, version 5.2.2(1)-release (x86_64-redhat-linux-gnu)
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

### PR DESCRIPTION
**Pull request by Eeems**
- [x] Rework makefile to use virtual environment
- [x] Hack in toltecmk to the build system and remove all unused old python code
- [x] Update github actions to cache the virtual environment pip install
- [x] Update to toltecmk 0.3.0 and `builder.make(..., False)`
- [x] Clean up implementation to simplify hacked in logic

### CODE REVIEW

**Comment by Eeems** [comment]
Currently waiting on https://github.com/toltec-dev/build/pull/39 and toltecmk 0.3.0 to be release.

**Comment by Eeems** [comment]
That should be cleaned up enough for now.

**Comment by Eeems** [comment]
Blocked waiting on https://github.com/toltec-dev/build/pull/49 and a new release.

**Comment by Eeems** [comment]
Opened #866 to resolve micro no longer building.

**Comment by Eeems** [comment]
Opened #867 to resolve mmc-utils no longer building.

**Comment by Eeems** [comment]
- netsurf fails to build. Opened #871 as it seems to be pre-existing.
- ~~webinterface-upload-button's sha256sum no longer matches~~
- I still need to test appmarkable, but github is having some issues that makes it so I can't download libfreetype at the moment.
  - I've asked @LinusCDE to take a look. I think this might be an issue with it's build system and not the network error it says it is.
  - #875

File: scripts/build/repo.py [hunk 1]
```diff
@@ -3,21 +3,36 @@
...
 import logging
 import os
+import pathlib
 import shutil
-import textwrap
-from typing import Dict, Iterable, List, Optional, Set
+
+from datetime import datetime
+from enum import auto
+from enum import Enum
```
Comment (Jayy001) [inline scripts/build/repo.py:13]:
```
from enum import auto, Enum
```

Comment (Eeems) [reply scripts/build/repo.py:13]: That makes diffs have more noise when changing it in the future. It's also not being flagged as an issue by our linting, so it's not really something we care about in the reviews.

**Review by Jayy001 — APPROVED** [review]

**Review by raisjn — APPROVED** [review]
lets do it

**Pull request merged**